### PR TITLE
feat/fullscreen migration

### PR DESCRIPTION
**Pull request by talevy17**

### CODE REVIEW

**Comment by talevy17** [comment]
Basically ready for initial release, should prioritise updating pro-gallery to latest in fullscreen and plugin-gallery

File: packages/fullScreen/web/src/FullScreen.jsx [hunk 1]
```diff
@@ -1,114 +1,75 @@
...
-                HeaderClose: addChildTo(carouselComponents.HeaderClose, CloseIcon),
-                HeaderFullscreen,
-              }}
-              styles={this.styles}
-            />
-          </Modal>
-        ) : null}
-      </ModalGateway>
-    );
+  if (target) {
```
Comment (shaulgo) [inline packages/fullScreen/web/src/FullScreen.jsx]: test this

File: packages/fullScreen/web/src/fullscreen.rtlignore.scss [hunk 2]
```diff
@@ -0,0 +1,33 @@
+.fullscreen {
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100vw;
+  height: 100vh;
+  z-index: 1000;
+  background: rgba(0,0,0,1);
+  transition: opacity 2s ease;
```
Comment (shaulgo) [inline packages/fullScreen/web/src/fullscreen.rtlignore.scss]: remove if not used

File: packages/fullScreen/web/src/fullscreen.rtlignore.scss [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  visibility: visible;
```
Comment (shaulgo) [inline packages/fullScreen/web/src/fullscreen.rtlignore.scss]: is this needed?

File: packages/fullScreen/web/src/fullscreen.rtlignore.scss [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  opacity: 1;
```
Comment (shaulgo) [inline packages/fullScreen/web/src/fullscreen.rtlignore.scss]: also ;)

File: packages/fullScreen/web/src/FullScreen.jsx [hunk 5]
```diff
@@ -1,114 +1,75 @@
...
 Fullscreen.propTypes = {
   images: PropTypes.array.isRequired,
-  index: PropTypes.number,
   isOpen: PropTypes.bool,
-  onClose: PropTypes.func.isRequired,
-  topMargin: PropTypes.number,
-  backgroundColor: PropTypes.string,
-  foregroundColor: PropTypes.string,
+  index: PropTypes.number,
+  topMargin: PropTypes.object,
```
Comment (shaulgo) [inline packages/fullScreen/web/src/FullScreen.jsx]: check topMargin and color props

Comment (talevy17) [reply packages/fullScreen/web/src/FullScreen.jsx]: works perfectly

File: packages/fullScreen/web/src/FullScreen.jsx [hunk 6]
```diff
@@ -1,114 +1,75 @@
...
-function addChildTo(Comp, Child) {
-  Comp.displayName = Comp.name;
-  return props => (
-    <Comp {...props}>
-      {/* eslint-disable-next-line react/prop-types */}
-      <Child {...props.innerProps} />
-    </Comp>
-  );
-}
+const { ProGallery } = process.env.SANTA ? {} : require('pro-gallery');
```
[the remark this answers is not in the document]
Comment (talevy17) [reply packages/fullScreen/web/src/FullScreen.jsx]: this change seems to break the gallery plugin for some reason, didn't figure out why though

File: packages/fullScreen/web/src/FullScreen.jsx (hunk 1 above)
Comment (talevy17) [reply packages/fullScreen/web/src/FullScreen.jsx]: tested, seems to be working fine

**Pull request merged**